### PR TITLE
Fix to driver history

### PR DESCRIPTION
**Pull request by dabail10**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
Fix from @cmbitz for Issue #420.
- [X] Developer(s): 
@dabail10 (D. Bailey)
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.

There are some missing baselines for PGI here.
MISS cheyenne_pgi_smoke_col_1x1_diag1_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_debug_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_bgcispol_debug compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_bgcnice_debug compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_bgcsklnice_debug compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_debug_run1year_thermo1 compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_debug_run1year_swredist compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_debug_run1year_swccsm3 compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_alt01_debug_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_alt02_debug_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_alt03_debug_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_alt04_debug_run1year compare icepack.8f96707a90.221210-095008 missing-data
MISS cheyenne_pgi_smoke_col_1x1_debug_dt30min_leap_run1year compare icepack.8f96707a90.221210-095008 missing-data
 MISS - comparison data is missing
13 of 495 tests MISSING data

https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_hash#8f96707a90132ca119d81ed84e5a62ca0ff3ed96
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

### CODE REVIEW

File: configuration/driver/icedrv_history.F90 [hunk 1]
```diff
@@ -364,17 +431,17 @@ subroutine history_write()
       start4(1) = 1
       count4(1) = nx
       start4(2) = 1
-      count4(2) = ncat
+      count4(2) = ntrcr
       start4(3) = 1
-      count4(3) = ntrcr
+      count4(3) = ncat
```
Comment (eclare108213) [inline configuration/driver/icedrv_history.F90:436]: does this change only impact FSD output?

**Review by eclare108213 — COMMENTED** [review]
Did you confirm that the history output for all variables except FSD remains the same as before?

File: configuration/driver/icedrv_history.F90 (hunk 1 above)
Comment (cmbitz) [reply configuration/driver/icedrv_history.F90:436]: It impacts all the variables in trcrn, which includes many other tracers. All of them are currently set to -9999 in the history file, even without the ITD.

**Comment by dabail10** [comment]
The log files were bfb.

**Comment by eclare108213** [comment]
> The log files were bfb.

Yes, but this changes the history output.  Does it look right?

**Comment by dabail10** [comment]
I will run an io_suite.

**Comment by dabail10** [comment]
I have confirmed that trcrn now has valid values where it had -9999 before.

**Review by eclare108213 — APPROVED** [review]
Excellent.  Thank you both, @cmbitz and @dabail10, for fixing this bug.

**Comment by apcraig** [comment]
Do we need a similar fix in CICE?

**Comment by dabail10** [comment]
I do not believe so. We do not dump the whole trcr/trcrn array in CICE like we do in Icepack.

**Comment by dabail10** [comment]
This should be good to go.

**Review by apcraig — APPROVED** [review]

**Pull request merged**